service/history/api/create_workflow_util.go

```diff
@@ -48,7 +48,7 @@ import (
 )
 
 type (
-	CreateWorkflowCASPredicate struct {
+	MutableStateSnapshot struct {
```

Comment:
Is it possible to call this something else?
Mutable state or workflow snapshot usually refer to the definition here: https://github.com/temporalio/temporal/blob/main/common/persistence/data_interfaces.go#L372

Reply:
@yycptt It is a mutable state snapshot. Unfortunately, I cannot come with the name which will describe what it is and not call it mutable state snapshot. We can rename it later if we find a better name. On the bright side, persistence is a different namespace, so there is no confusion.

Reply:
Maybe we have a different definition for Snapshot. I am thinking snapshot should contain all states in the mutable state. But this struct only contains two fields for the CAS operation in persistence layer. Looks like it has two purposes here, one for the cas operation and the other one for suggesting that a new workflow should be created.

Reply:
It's more a version than a snapshot as it contains no data. What about `MutableStateVersion`?

Reply:
@tdeebswihart There is also RunID. 
@yycptt I believe we have the same definition of snapshot. We memorize some current data for later use. But different namespaces can have different definitions of MutableState. Types should be defined in the context of the user. It is the user who defines what functionality the type should provide. For different users MutableState means different things. Persistance nees all the data required to restore MutabletSateImpl. But for the startwithsignalworkflow just two fields are enough.

Reply:
hmmm I guess I still fail to follow. I understand things can have different meaning in different packages. But RunID + Version is a snapshot for "mutable state" in api package is something new to me. My brain is still read them as additional information needed for creating the new workflow and is "CASPredicateSnapshot" now I guess. 

I don't want to block your work though, so I will leave the decision to the team.